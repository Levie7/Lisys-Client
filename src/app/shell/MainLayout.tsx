import { Layout } from 'antd';
import { enquireScreen, unenquireScreen } from 'enquire-js';
import React from 'react';

import { UIContext } from 'src/shared/contexts/UIContext';

import { Header } from './components/Header/Header';
import { Footer } from './components/Footer/Footer';
import { MainMenu } from './components/MainMenu/MainMenu';

interface MainLayoutState {
    isMobile: boolean;
}

export class MainLayout extends React.Component<{}, MainLayoutState> {
    enquireHandler: any;

    constructor(props: {}) {
        super(props);
        this.state = { isMobile: false };
    }

    componentDidMount() {
        this.enquireHandler = enquireScreen((mobile: any) => {
            this.setState({ isMobile: mobile });
        });
    }

    componentWillUnmount() {
        unenquireScreen(this.enquireHandler);
    }

    render() {
        let { children } = this.props;
        let UI = { isMobile: !!this.state.isMobile! };

        return (
            <UIContext.Provider value={UI}>
                <Layout>
                    <MainMenu />
                    <Layout>
                        <Layout.Header>
                            <Header />
                        </Layout.Header>
                        <Layout.Content>{children}</Layout.Content>
                        <Layout.Footer>
                            <Footer />
                        </Layout.Footer>
                    </Layout>
                </Layout>
            </UIContext.Provider>
        );
    }
}
