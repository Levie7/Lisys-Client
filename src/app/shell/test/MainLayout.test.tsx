import { shallow } from 'enzyme';
import * as React from 'react';

import { MainLayout } from '../MainLayout';

describe('MainLayout', () => {
    const wrap = shallow(<MainLayout />);
    wrap.setState({ isMobile: false });

    const setState = jest.fn().mockImplementation((mobile: boolean) => {
        wrap.setState({ isMobile: mobile });
    });
    const enquireScreen = jest.fn().mockImplementation((mobile: any) => {
        setState(mobile);
    });

    it('<MainLayout/> - should render mainLayout component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should contains all children component', () => {
        expect(wrap.find('ContextProvider').exists()).toBeTruthy();
        expect(wrap.find('Layout').exists()).toBeTruthy();
        expect(wrap.find('Header').exists()).toBeTruthy();
        expect(wrap.find('Content').exists()).toBeTruthy();
        expect(wrap.find('Footer').exists()).toBeTruthy();
        expect(wrap.find('Memo()').exists()).toBeTruthy();
    });

    it('should call component did mount', () => {
        MainLayout.prototype.componentDidMount();
    });

    describe('when enquireHandler called', () => {
        const mobile = true;
        let enquireHandler: any;
        beforeEach(() => {
            enquireHandler = jest
                .spyOn(wrap.instance(), 'enquireHandler')
                .mockImplementation(() => {
                    enquireScreen(mobile);
                });
            wrap.instance().enquireHandler();
        });

        it('should call enquireScreen', () => {
            expect(enquireHandler).toHaveBeenCalled();
            expect(enquireScreen).toHaveBeenCalled();
            expect(setState).toHaveBeenCalled();
        });

        it('should set state depend on is mobile', () => {
            expect(wrap.state('isMobile')).toBeTruthy();
        });
    });

    it('should have property unmount', () => {
        wrap.unmount();
    });
});
