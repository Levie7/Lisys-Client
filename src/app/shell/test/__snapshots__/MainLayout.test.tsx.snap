// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout <MainLayout/> - should render mainLayout component correctly 1`] = `ShallowWrapper {}`;
